CRM/Utils/Type.php

```diff
@@ -470,18 +435,6 @@ public static function validate($data, $type, $abort = TRUE, $name = 'One of par
         return $data;
 
       case 'Date':
-        // a null date is valid
-        if (strlen(trim($data)) == 0) {
-          return trim($data);
-        }
-
-        if (preg_match('/^\d{8}$/', $data) &&
-          CRM_Utils_Rule::mysqlDate($data)
```

Comment:
Isn't this is a functional difference, since 'Date' only accepted date input, and Timestamp accepts Date+Time?

Reply:
@mlutfy Yes you are right.  Looking in core it seems the validate(xx, 'Date') is not really used, and extensions have been using "Integer" to validate DateTime. https://github.com/eileenmcnaughton/nz.co.fuzion.extendedreport/blob/master/CRM/Extendedreport/Form/Report/Campaign/CampaignProgressReport.php#L144

For `escape` we accept both.  So this change would allow `Date` to validate a DateTime as well which I think it should have been doing anyway?  @seamuslee001 @eileenmcnaughton @pfigel What are your thoughts on this point?

Reply:
@mattwire I agree that being consistent in `validate` and `escape` would make sense.

I found some usages of `CRM_Utils_Type::validate()` with `$type='Date'` in universe, e.g. in `uk.org.futurefirst.networks.civipoints/CRM/Points/Page/AJAX.php`. I don't believe the change in behaviour would really cause a serious issue in this case (worst case I imagine MySQL would truncate the time portion while potentially emitting a warning). There should at least be no security impact.